internal/metrics/metrics.go

```diff
@@ -6,31 +6,44 @@ import (
 	"github.com/prometheus/client_golang/prometheus/promauto"
 )
 
-// TODO Update metric names according to: https://prometheus.io/docs/instrumenting/writing_exporters/#naming
 const (
-	NameSpace           = "idmsvc"
-	HttpStatusHistogram = "http_status_histogram"
+	// NameSpace is the namespace for the Prometheus metrics
+	NameSpace = "idmsvc"
 )
 
+// Metrics holds all the Prometheus metrics for the application
 type Metrics struct {
-	HttpStatusHistogram prometheus.HistogramVec
+	// HTTPRequestDuration is a histogram that measures the duration of HTTP requests
+	HTTPRequestDuration *prometheus.HistogramVec
+	// HttpRequestsTotal is a counter that measures the total number of HTTP requests
+	HTTPRequestsTotal *prometheus.CounterVec
```

Comment:
I think the second metric `HTTPRequestTotal` would not be necessary; I have launched some local requests, and I get the below (removing the `_bucket` values for short):

> See `idmsvc_http_request_duration_seconds_count` and `idmsvc_http_requests_total`
> the values are matching.

Are there any difference between `idmsvc_http_request_duration_seconds_count` and `idmsvc_http_requests_total` values?

```raw
# HELP go_build_info Build information about the main Go module.
# TYPE go_build_info gauge
go_build_info{checksum="",path="github.com/podengo-project/idmsvc-backend",version="(devel)"} 1
# HELP idmsvc_http_request_duration_seconds Duration of HTTP requests
# TYPE idmsvc_http_request_duration_seconds histogram

idmsvc_http_request_duration_seconds_sum{method="POST",path="/api/idmsvc/v1/domains",status="201"} 0.017798608
idmsvc_http_request_duration_seconds_count{method="POST",path="/api/idmsvc/v1/domains",status="201"} 1

idmsvc_http_request_duration_seconds_sum{method="POST",path="/api/idmsvc/v1/domains",status="409"} 0.03447769900000001
idmsvc_http_request_duration_seconds_count{method="POST",path="/api/idmsvc/v1/domains",status="409"} 13

idmsvc_http_request_duration_seconds_sum{method="POST",path="/api/idmsvc/v1/domains/token",status="200"} 0.010131954999999998
idmsvc_http_request_duration_seconds_count{method="POST",path="/api/idmsvc/v1/domains/token",status="200"} 10
# HELP idmsvc_http_requests_total Total number of HTTP requests
# TYPE idmsvc_http_requests_total counter
idmsvc_http_requests_total{method="POST",path="/api/idmsvc/v1/domains",status="201"} 1
idmsvc_http_requests_total{method="POST",path="/api/idmsvc/v1/domains",status="409"} 13
idmsvc_http_requests_total{method="POST",path="/api/idmsvc/v1/domains/token",status="200"} 10
# HELP promhttp_metric_handler_errors_total Total number of internal errors encountered by the promhttp metric handler.
# TYPE promhttp_metric_handler_errors_total counter
promhttp_metric_handler_errors_total{cause="encoding"} 0
promhttp_metric_handler_errors_total{cause="gathering"} 0
```

Reply:
Yes, that's what I mentioned on the meeting on Monday. I was thinking of removing it but kept it for discussion.

Reply:
I've removed the `http_requests_total` + the no longer used `mapStatus` function.